### PR TITLE
Towards vanilla annotator

### PR DESCRIPTION
**Pull request by nickstenning**
A bunch of changes that bring `h/static/scripts/annotator/` closer in line with the upstream Annotator v1.2.x branch.

### CODE REVIEW

File: h/assets.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ annotator:
     - h:static/scripts/annotator/editor.coffee
     - h:static/scripts/annotator/viewer.coffee
 
+    # Monkeypatch range.coffee
+    - h:static/scripts/annotator/range_monkey.coffee
```
Comment (aron) [inline h/assets.yaml]: :rage1: :monkey_face:

File: h/static/scripts/annotator/range.coffee [hunk 2]
```diff
@@ -493,8 +450,8 @@ class Range.SerializedRange
   # Public: Returns the range as an Object literal.
   toObject: ->
     {
-      startContainer: @startContainer
+      start: @start
       startOffset: @startOffset
-      endContainer: @endContainer
+      end: @end
```
Comment (aron) [inline h/static/scripts/annotator/range.coffee]: Is this not going to break existing annotations? I can't see a transform function.

Comment (tilgovi) [inline h/static/scripts/annotator/range.coffee]: I think that's range_monkey

Comment (tilgovi) [inline h/static/scripts/annotator/range.coffee]: Oh, I see. @nickstenning I think we need range_monkey to patch sniff.

**Comment by aron** [comment]
Solid :golf:

File: h/static/scripts/annotator/annotator.coffee [hunk 3]
```diff
@@ -329,6 +297,7 @@ class Annotator extends Delegator
   # Returns annotation Object.
   updateAnnotation: (annotation) ->
     this.publish('beforeAnnotationUpdated', [annotation])
+    $(annotation.highlights).attr('data-annotation-id', annotation.id)
```
Comment (csillag) [inline h/static/scripts/annotator/annotator.coffee]: `annotation.highlights` is not a thing on our branch.
Instead, we have `annotation.anchors`, which is an array, and then each element has a `highlight` field, which is a map, for pageNumber -> highlight node.

Comment (csillag) [inline h/static/scripts/annotator/annotator.coffee]: Although it's harmless (just does not do anything)

Comment (nickstenning) [inline h/static/scripts/annotator/annotator.coffee]: > Although it's harmless (just does not do anything)

:+1:

**Comment by csillag** [comment]
LGTM.

**Comment by gergely-ujvari** [comment]
LGTM

**Pull request merged**